chore(sidebarwidgets): refresh styles

dancormier opened the pull request:
[STACKS-673](https://stackoverflow.atlassian.net/browse/STACKS-673)

This PR refreshes styles for the Stacks Sidebar Widget component, per the [Figma](https://www.figma.com/design/CGU7JFDfnKDtX8QXCm1I2f/Sidebar-Widgets?node-id=1-2&node-type=canvas&t=10krLqtDcZCwjKmG-0).

## How to test

> [!TIP]
> ~~Netlify deploy previews are not working right now, so you'll need to run this locally to see the changes~~ 
> **[Dev preview](https://6723abd5b8988b439233e329--stacks.netlify.app/)**

1. Visit the **[Dev preview](https://6723abd5b8988b439233e329--stacks.netlify.app/product/components/sidebar-widgets/)**
    - To test locally, checkout out this branch (`STACKS-673/sidebarwidget-style-refresh`) locally and run it (`npm i && npm run start`), head to [the sidebar widgets docs page](http://localhost:8080/product/components/sidebar-widgets/) to see the changes

## Screenshots

### Basic

<details>
<summary>Before</summary>

![image](https://github.com/user-attachments/assets/d6f85660-e376-4f03-938b-f9e131a28473)

</details>

<details>
<summary>After</summary>

![image](https://github.com/user-attachments/assets/3dc6d311-7fbc-441f-b0a8-532e8a2492cb)

</details>


### Navigation

<details>
<summary>Before</summary>

![image](https://github.com/user-attachments/assets/9714c7da-3a57-46b6-84d2-9c4b3f90ed67)

</details>

<details>
<summary>After</summary>

![image](https://github.com/user-attachments/assets/046839ae-5dcb-4adc-95d0-2e2d4c39e57d)

</details>

### Header actions

<details>
<summary>Before</summary>

![image](https://github.com/user-attachments/assets/27007e52-71ed-49e9-8101-2df0e93e52ae)

</details>

<details>
<summary>After</summary>

![image](https://github.com/user-attachments/assets/9ac43c99-e104-4dd9-9bd6-bfa5b03d6f08)

</details>

### Accordion

<details>
<summary>Before</summary>

![image](https://github.com/user-attachments/assets/26d5ceda-a7ab-4290-a564-20e6e8cac74d)

</details>

<details>
<summary>After</summary>

![image](https://github.com/user-attachments/assets/d81a0089-9a24-4f7d-a8c0-64cbaf045b39)

</details>

dancormier committed:
chore(sidebarwidgets): refresh styles

[STACKS-673](https://stackoverflow.atlassian.net/browse/STACKS-673)

dancormier committed:
add a11y; visual tests

dancormier committed:
Merge branch 'develop' into STACKS-673/sidebarwidget-style-refresh

dancormier committed:
Fix formatting

dancormier committed:
Merge branch 'develop' into STACKS-673/sidebarwidget-style-refresh

dancormier committed:
Fix visited tag font color within widget

giamir commented on screenshots/Chromium/baseline/s-sidebarwidget-dark-blue-ws3-small-bold-headers.png:
`small-bold-headers` screenshots look odd. We might need to add a `s-sidebarwidget--content` sibling element in the test setup because the header element by itself behave strangely not leaving enough padding at the bottom. This is probably because you are not supposed to use a header element in isolation.

giamir reviewed (approved):
Great work @dancormier. Code changes make sense to me. I found only a small issue with a variant of the visual tests (see comment).

giamir commented:
> Netlify deploy previews are not working right now, so you'll need to run this locally to see the changes

@dancormier I have spent some time trying to fix this but unfortunately it doesn't look netlify is honoring the `GIT_LFS_SKIP_SMUDGE` variable nor the `GIT_LFS_FETCH_INCLUDE` which they should according to [their docs](https://docs.netlify.com/configure-builds/environment-variables/#netlify-configuration-variables). 

I talked to @abovedave and as a first step we can reach out to them for support via email. 
Worst case scenario we can move to use their CLI for direct deployments in github workflows (I did a test and it works) instead of using the GH repo linked integration. The downside of doing that is that we won't have the netlify bot commenting on PRs, etc... We could achieve something similar also by expanding the GH workflow we would create but it would be nice if we could just stick to their linked repo option. I will draft a support email and cc you.

dancormier committed:
Update small bold text test

Addresses https://github.com/StackExchange/Stacks/pull/1841#discussion_r1820479567

CGuindon reviewed (approved):
LGTM but would still wait for a check from @andrewmeacham

andrewmeacham reviewed (approved):
Looks great!
